docs/firmware_flashing.md

```diff
@@ -130,4 +130,112 @@ Then run following commad:
 grub-mkconfig -o /boot/grub/grub.cfg
 ```
 
-Finally reboot
\ No newline at end of file
+Finally reboot
+
+APU firmware updater for OPNsense
+----------------
+
+Login via SSH to your OPNsense firewall and place the following script
+where you want it:
+
+```sh
+#!/bin/sh
+###
+# APU firmware updater for OPNsense
+#
+# Warning:
+# If you want to use this script for APU1, add the param
+# `-c "MX25L1605A/MX25L1606E/MX25L1608E"`
+# to the line with the `flashrom` command.
+#
+# Requires:
+# - flashrom (pkg install -y flashrom)
+#
+# Installation:
+# - Login via SSH to the firewall
+# - Create the script where you want it
+# - Make it executable (chmod +x firmware-updater.sh)
+###
+
+TYPE="apu2"
+VERSION="4.12.0.4"
+
+SRC="https://3mdeb.com/open-source-firmware/pcengines/${TYPE}/"
+PREFIX="${TYPE}_v${VERSION}"
+FILE="${PREFIX}.rom"
+CHECKSUM="${PREFIX}.SHA256"
+
+echo
+echo "+-----------------------------------+"
+echo "| APU firmware updater for OPNsense |"
+echo "+-----------------------------------+"
+
+cd "/tmp"
+
+echo
+echo "Settings"
+echo "- Type: ${TYPE}"
+echo "- Version: ${VERSION}"
+
+echo
+echo "Downloading files ..."
+
+echo "- ${FILE}"
+curl -sS "${SRC}${FILE}" -o "${FILE}"
+
+echo "- ${CHECKSUM}"
+curl -sS "${SRC}${CHECKSUM}" -o "${CHECKSUM}"
+
+if [ $? != 0 ]; then
+  echo "... failed."
+
+  echo
+  exit 1
+fi
+
+echo "... done."
+
+echo
+echo "Verify checksum ..."
+
+shasum -c "${CHECKSUM}"
+
+if [ $? != 0 ]; then
+  echo "... failed."
+
+  echo
+  exit 1
+fi
+
+echo "... success."
+
+echo
+echo "Flash firmware ..."
+
+flashrom -w "${FILE}" -p internal
```

Comment:
@lwcislo you may search for a way to detect platform name and automatically append the additional flashrom paratmers. Maybe sysctl variables hold the platform name?

Reply:
```sh
/usr/local/sbin/dmidecode -q -t system | grep ':' | tr -d '\t' | awk -F: '{print $1" ="$2}' | grep 'Manufacturer\|Product\|Version\|Serial\|Family'

Manufacturer = PC Engines
Product Name = apu2
Version = 1.0
Serial Number = 1150058
Family = Not Specified
```

```sh
/usr/local/sbin/dmidecode -q -t bios | grep ':' | tr -d '\t' | awk -F: '{print $1" ="$2}' | grep 'Vendor\|Version\|Release'

Vendor = coreboot
Version = v4.12.0.4
Release Date = 08/27/2020
```

Reply:
```sh
/usr/local/sbin/dmidecode -s "system-product-name"

apu2
```

But not sure if this will work for old versions of BIOS?

Reply:
This will probably fail when the Product name was broken, i.e. contained the Manufacturer and was written upper case.

Reply:
Our best bet would be the user knowing what he's doing.

As it affects only `APU1` and not `APU2/3/4/5` we may read the number from the param `TYPE` directly?

Reply:
Right, we may realy on input parameters.. Cool. Looks good.

Reply:
Verified, looks good.